OneSignalSDK/onesignal/src/main/java/com/onesignal/OSInAppMessageController.java

```diff
@@ -27,16 +29,20 @@
     }};
 
     public static final String IN_APP_MESSAGES_JSON_KEY = "in_app_messages";
+    private static final String OS_SAVE_IN_APP_MESSAGE = "OS_SAVE_IN_APP_MESSAGE";
 
     OSTriggerController triggerController;
     private OSSystemConditionController systemConditionController;
+    private OSInAppMessageRepository inAppMessageRepository;
 
     // IAMs loaded remotely from on_session
     //   If on_session won't be called this will be loaded from cache
     @NonNull private ArrayList<OSInAppMessage> messages;
     // IAMs that have had their trigger(s) evaluated to true;
     //   This mean they have been added to the queue to display, or have already displayed
-    @NonNull final private Set<String> triggeredMessages;
+    @NonNull final private Set<String> dismissedAndToDismissMessages;
+    // IAMs displayed with last displayed time and quantity of displays data
+    @NonNull final private List<OSInAppMessage> displayedMessagesData;
```

Comment:
Why aren't we using the previous `messages` instead?

Reply:
do you mean ArrayList<OSInAppMessage> messages?
i think is more clear to have separate responsibilities. But i can mix the logic to whenever the message is set add the data for redisplay,  although it will trigger another for loop. We can discuss this if you want

Reply:
I am wondering why we have
`@NonNull private ArrayList<OSInAppMessage> messages;`
and
`@NonNull final private List<OSInAppMessage> displayedMessagesData;`

Shouldn't we just have one list of messages that we obtain from the `on_session` or cached IAMs?

Reply:
yeah i can mix the logics

Reply:
I think we should have a single `@NonNull private ArrayList<OSInAppMessage> messages;` that contains on IAMs pulled down for the current session

Reply:
@mikechoch and I talked about this. Our `ArrayList<OSInAppMessage> messages` is a list of all IAMs from our server so we should keep this list untouched.

To read and write our two new properties I think we should create a new class to store them under. We can call this something like `OSInAppMessageDisplayStats` which will hold our `displayQuantity` and `lastTimeDisplayed`. This can still be referenced by `OSInAppMessage` as a property.

Reply:
ok that make sense. I agree with that the message list should kept untouched.